packages/renderer/src/lib/ui/Typeahead.svelte

```diff
@@ -1,33 +1,55 @@
 <script lang="ts">
 import { Spinner } from '@podman-desktop/ui-svelte';
 
-type SearchFunction = (s: string) => Promise<string[]>;
-
-// the text displayed when no option is selected
-export let placeholder: string | undefined = undefined;
-export let required: boolean = false;
-export let delay: number = 200;
-export let disabled: boolean = false;
-export let initialFocus: boolean = false;
-export let id: string | undefined = undefined;
-export let name: string | undefined = undefined;
-export let error: boolean = false;
-
-export let searchFunction: SearchFunction = async (_s: string) => [];
-export let onChange = function (_s: string) {};
-export let onEnter = function () {};
+interface GroupItem {
+  values: string[];
+  group?: string;
+  sorted?: boolean;
+}
+
+interface Props {
+  placeholder?: string;
+  required?: boolean;
+  delay?: number;
+  disabled?: boolean;
+  initialFocus?: boolean;
+  id?: string;
+  name?: string;
+  error?: boolean;
+  className?: string;
+  onInputChange?: (s: string) => Promise<GroupItem[]>;
+  onChange?: (s: string) => void;
+  onEnter?: () => void;
+}
+
+// the placeholder text displayed when no option is selected
+let {
+  placeholder,
+  required = false,
+  delay = 200,
+  disabled = false,
+  initialFocus = false,
+  id,
+  name,
+  error = false,
+  className,
+  onInputChange = async (_s: string) => [],
+  onChange = function (_s: string) {},
+  onEnter = function () {},
```

Comment:
Probably better to keep them undefined, and use conditional calling.

E.g.

```ts
onChange: (() => void) | undefined;

...
// Conditional calling
onChange?.();
```

Reply:
This can be a future PR, don't feel forced to include it, as this was already the case before, I don't mind merging it.